Lacework/lacework_module/lacework_connector.py

```diff
@@ -0,0 +1,171 @@
+                start = time.time()
+
+                try:
+                    self.forward_next_batches()
+                except (HTTPError, BaseHTTPError) as ex:
+                    self.log_exception(ex, message="Failed to get next batch of events")
+                except Exception as ex:
+                    self.log_exception(ex, message="An unknown exception occurred")
+                    raise
+
+                # compute the duration of the last events fetching
+                duration = int(time.time() - start)
+                FORWARD_EVENTS_DURATION.labels(intake_key=self.configuration.intake_key).observe(duration)
+
+                # Compute the remaining sleeping time
+                delta_sleep = self.configuration.frequency - duration
+                # if greater than 0, sleep
+                if delta_sleep > 0:
+                    time.sleep(delta_sleep)
+        finally:
+            self.log(message="Lacework Events Trigger has stopped", level="info")
+
+    def get_next_events(self, cursor: str | None) -> dict | None:
+        # set parameters
+        parameters = {
+            "limit": self.pagination_limit,
+        }
+
+        # if defined, set the cursor
+        if cursor:
+            parameters["cursor"] = cursor
+        # otherwise, get events starting from the last 5 minutes
+        else:
+            parameters["startTime"] = int(time.time()) - 300
+
+        # Get the events
+        response = self.client.list_alerts(parameters=parameters)
+
+        # Something failed
+        if not response.ok:
+            self.log(
+                message=(
+                    "Request on Lacework Central API to fetch events of failed with"
+                    f" status {response.status_code} - {response.reason}"
+                ),
+                level="error",
+            )
+
+            return None
+
+        return response.json()
+
+    def _get_most_recent_timestamp_from_items(self, items: list[dict]):
+        def _extract_timestamp(item: dict) -> float:
+            RFC3339_STRICT_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+            return datetime.datetime.strptime(item["startTime"], RFC3339_STRICT_FORMAT).timestamp()
+
+        latest_message = max(items, key=lambda item: item["startTime"])
+        latest_message_timestamp = _extract_timestamp(latest_message)
+        return latest_message_timestamp
+
+    def forward_next_batches(self) -> None:
+        """
+        Successively queries the Lacework Central API while more are available
+        and the current batch is not too big.
+        """
+        has_more_messages = True
+        cursor = self.cursor
+
+        messages = []
+        while has_more_messages and self.running:
+            has_more_messages = False
+
+            batch = self.get_next_events(cursor)
+            if batch is None:
+                break
+
+            if batch.content.paging.urls.get("nextPage") != None:
+                has_more_messages = True,
```

Comment:
Be careful. With the comma at the end of the expression, Python will consider the value as a tuple.
```suggestion
                has_more_messages = True
```